python/packages/ai/teams/app.py

```diff
@@ -355,6 +359,130 @@ def __call__(func: RouteHandler[StateT]) -> RouteHandler[StateT]:
 
         return __call__
 
+    def file_consent_accept(
+        self,
+    ) -> Callable[
+        [Callable[[TurnContext, StateT, FileConsentCardResponse], Awaitable[None]]],
+        Callable[[TurnContext, StateT, FileConsentCardResponse], Awaitable[None]],
+    ]:
+        """
+        Registers a handler for when a file consent card is accepted by the user.
+         ```python
+        # Use this method as a decorator
+        @app.file_consent_accept()
+        async def on_file_consent_accept(
+            context: TurnContext, state: TurnState, file_consent_response: FileConsentCardResponse
+        ):
+            print(file_consent_response)
+        # Pass a function to this method
+        app.file_consent_accept()(on_file_consent_accept)
+        ```
+        """
+
+        def __selector__(context: TurnContext) -> bool:
+            return (
+                context.activity.type == ActivityTypes.invoke
+                and context.activity.name == "fileConsent/invoke"
+                and isinstance(context.activity.value, dict)
+                and context.activity.value.get("action") == "accept"
+            )
+
+        def __call__(
+            func: Callable[[TurnContext, StateT, FileConsentCardResponse], Awaitable[None]]
+        ) -> Callable[[TurnContext, StateT, FileConsentCardResponse], Awaitable[None]]:
+            async def __handler__(context: TurnContext, state: StateT):
+                if not context.activity.value:
+                    return False
+                await func(context, state, context.activity.value)
+                return True
+
+            self._routes.append(Route[StateT](__selector__, __handler__))
```

Comment:
I see the activity type is invoke, should set `is_invoke` parameter to `True` in `Route`'s constructor.